modules/context/package.go

```diff
@@ -83,12 +83,15 @@ func packageAssignment(ctx *Context, errCb func(int, string, interface{})) {
 }
 
 func determineAccessMode(ctx *Context) (perm.AccessMode, error) {
-	accessMode := perm.AccessModeNone
-
 	if setting.Service.RequireSignInView && ctx.Doer == nil {
-		return accessMode, nil
+		return perm.AccessModeNone, nil
 	}
 
+	if ctx.Doer != nil && !ctx.Doer.IsGhost() && (!ctx.Doer.IsActive || ctx.Doer.ProhibitLogin) {
```

Comment:
How about " ctx.Doer != nil && ctx.Doer.IsGhost()", should a ghost user also considered as none access?

Reply:
No, the ghost user is used by the container registry. Even anonymous downloads need a login by the spec and I used the ghost user for that. So the ghost user has read access.